tests/python/unittest/test_operator.py

```diff
@@ -2515,6 +2515,20 @@ def test_slice_like():
             assert_allclose(xx, xgrad.asnumpy())
             assert_allclose(xgrad1.asnumpy(), mx.nd.zeros_like(xgrad1).asnumpy())
 
+@with_seed()
+def test_slice_like_different_types():
+    x = [[  1.,   2.,   3.,   4.],
```

Comment:
Though this example looks good, I am afraid it is specific. Do you mind using randomized shape, size and NDarray initalization to ensure this is generic?

Reply:
Those parameters do not matter for this test (and they are already tested in the dedicated test of `slice_like` op functionality) - the values for `x` and `y` were taken from the example from `slice_like` documentation. The only thing that matters is the type of `y` in this test. Before this PR this code would not work - it would throw an exception during type inference. With this PR, this code works (as it should).

Reply:
Alright. Makes sense. Thanks for the explanation.